File: CarSpecAPI/Data/Models/DataExtractionModel/ImportBatchListDto.cs
namespace CarSpecAPI.Data.Models.DataExtractionModel
{
    public class ImportBatchListDto
    {
        public int ImportBatchId { get; set; }

        public DateTime StartedAt { get; set; }

        public DateTime? CompletedAt { get; set; }

        public string Status { get; set; } = string.Empty;

        public int? ImportedBy { get; set; }

        public string? Notes { get; set; }

        public string? AIModel { get; set; }

        public DateTime? AIProcessedAt { get; set; }

        public bool HasAIResult { get; set; }

        public bool HasFinalResult { get; set; }

        public List<ImportBatchDocumentDto> Documents { get; set; } = new();
    }
}

File: CarSpecAngular/carspec/src/app/core/models/interfaces/DataProcessingDtos/import-batch-list-dto.ts
import { ImportBatchDocumentDto } from "./import-batch-document-dto";

export interface ImportBatchListDto {
  importBatchId: number;

  startedAt: string;

  completedAt: string | null;

  status: string;

  importedBy: number | null;

  notes: string | null;

  aiModel: string | null;

  aiProcessedAt: string | null;

  hasAIResult: boolean;

  hasFinalResult: boolean;

  documents: ImportBatchDocumentDto[];
}
